ibm/service/codeengine/resource_ibm_code_engine_config_map.go

```diff
@@ -0,0 +1,292 @@
+
+	parts, err := flex.SepIdParts(d.Id(), "/")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getConfigMapOptions.SetProjectID(parts[0])
+	getConfigMapOptions.SetName(parts[1])
+
+	configMap, response, err := codeEngineClient.GetConfigMapWithContext(context, getConfigMapOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		log.Printf("[DEBUG] GetConfigMapWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("GetConfigMapWithContext failed %s\n%s", err, response))
+	}
+
+	if err = d.Set("project_id", configMap.ProjectID); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting project_id: %s", err))
+	}
+	if err = d.Set("name", configMap.Name); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting name: %s", err))
+	}
+	if !core.IsNil(configMap.Data) {
+		data := make(map[string]string)
+		for k, v := range configMap.Data {
+			data[k] = string(v)
+		}
+		if err = d.Set("data", data); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting data: %s", err))
+		}
+	}
+	if !core.IsNil(configMap.CreatedAt) {
+		if err = d.Set("created_at", configMap.CreatedAt); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting created_at: %s", err))
+		}
+	}
+	if err = d.Set("entity_tag", configMap.EntityTag); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting entity_tag: %s", err))
+	}
+	if !core.IsNil(configMap.Href) {
+		if err = d.Set("href", configMap.Href); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting href: %s", err))
+		}
+	}
+	if !core.IsNil(configMap.ID) {
+		if err = d.Set("id", configMap.ID); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting id: %s", err))
+		}
+	}
+	if !core.IsNil(configMap.ResourceType) {
+		if err = d.Set("resource_type", configMap.ResourceType); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting resource_type: %s", err))
+		}
+	}
+	if !core.IsNil(configMap.ID) {
+		if err = d.Set("config_map_id", configMap.ID); err != nil {
+			return diag.FromErr(fmt.Errorf("Error setting config_map_id: %s", err))
+		}
+	}
+	if err = d.Set("etag", response.Headers.Get("Etag")); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting etag: %s", err))
+	}
+
+	return nil
+}
+
+func resourceIbmCodeEngineConfigMapUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	codeEngineClient, err := meta.(conns.ClientSession).CodeEngineV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	replaceConfigMapOptions := &codeenginev2.ReplaceConfigMapOptions{}
+
+	parts, err := flex.SepIdParts(d.Id(), "/")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	replaceConfigMapOptions.SetProjectID(parts[0])
+	replaceConfigMapOptions.SetName(parts[1])
+
+	hasChange := false
+
+	if d.HasChange("project_id") {
+		return diag.FromErr(fmt.Errorf("Cannot update resource property \"%s\" with the ForceNew annotation."+
```

Comment:
This check is redundant, As forceNew argument in schema takes care of this check. :)

Reply:
Removed the redundant check.